### PR TITLE
adding Helicopter class with methods

### PR DESCRIPTION
**Pull request by smykandriy**

### CODE REVIEW

File: helicopter.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+        self.fuel_capacity = fuel_capacity
+        self.current_fuel = current_fuel
+
+    def __str__(self):
+        return f"""
+        Helicopter {self.id} ({self.model})
+        Max Altitude: {self.max_altitude}
+        Current Altitude: {self.current_altitude}
+        Fuel Capacity: {self.fuel_capacity}
+        Current fuel: {self.current_fuel}"""
```
Comment (MarkoYaminsky) [inline helicopter.py]: Make this with self.\_\_dict\_\_ and loop

File: helicopter.py [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+        if altitude < 0:
+            raise ValueError("Invalid altitude value: altitude cannot be negative.")
+        if self.current_altitude + altitude > self.max_altitude:
+            self.current_altitude = self.max_altitude
+            return
+        self.current_altitude += altitude
+
+    def descend(self, altitude):
+        if altitude < 0:
+            raise ValueError("Invalid altitude value: altitude cannot be negative.")
```
Comment (MarkoYaminsky) [inline helicopter.py]: same ValueError with exact same message is raised twice. You can make a function or even inherit from Exception and make this exact Error (i.e. HelicopterNegativeAltitudeError) without specifying message or with this message as default arg

**Review by MarkoYaminsky — APPROVED** [review]
good as for lab. But not everything is perfectly clean

File: helicopter.py [hunk 3]
```diff
@@ -0,0 +1,228 @@
...
+        """
+        attrs = {
+            "id": self.__id,
+            "model": self.__model,
+            "max_altitude": self.__max_altitude,
+            "current_altitude": self.__current_altitude,
+            "fuel_capacity": self.__fuel_capacity,
+            "current_fuel": self.__current_fuel,
+        }
+        docstring = [f"{key.capitalize()}: {attrs[key]}" for key in attrs]
```
Comment (MarkoYaminsky) [inline helicopter.py]: This is not docstring, this is attribute dictionary. Who is the one, that originated this idea? I see this 3rd time today

File: helicopter.py [hunk 4]
```diff
@@ -0,0 +1,228 @@
...
+        and current fuel level.
+        """
+        attrs = {
+            "id": self.__id,
+            "model": self.__model,
+            "max_altitude": self.__max_altitude,
+            "current_altitude": self.__current_altitude,
+            "fuel_capacity": self.__fuel_capacity,
+            "current_fuel": self.__current_fuel,
+        }
```
Comment (MarkoYaminsky) [inline helicopter.py]: Use \_\_dict\_\_

File: helicopter.py [hunk 5]
```diff
@@ -0,0 +1,228 @@
...
+if __name__ == "__main__":
+    helicopters = [
+        Helicopter(),
+        Helicopter(103, "Apache", 1200, 3000, 500, 430),
+        Helicopter.get_instance(),
+        Helicopter.get_instance(),
+    ]
+
+    for _ in helicopters:
+        print(_)
```
Comment (MarkoYaminsky) [inline helicopter.py]: Андрій, я тебе прибʼю за цей код в файлі з класом, а не в мейні

**Review by MarkoYaminsky — CHANGES_REQUESTED** [review]

File: helicopter.py (hunk 5 above)
Comment (smykandriy) [reply helicopter.py]: пан Яцук такий приклад показував на лекції, я зробив рівно так само

File: helicopter.py [hunk 6]
```diff
@@ -73,16 +73,8 @@ def __str__(self):
...
-            "id": self.__id,
-            "model": self.__model,
-            "max_altitude": self.__max_altitude,
-            "current_altitude": self.__current_altitude,
-            "fuel_capacity": self.__fuel_capacity,
-            "current_fuel": self.__current_fuel,
-        }
-        docstring = [f"{key.capitalize()}: {attrs[key]}" for key in attrs]
-        return f"{self.__class__.__name__}({', '.join(docstring)})"
+        attrs = [f"{key.split(sep='_')[-1]}: {self.__dict__[key]}" for key in self.__dict__]
```
Comment (MarkoYaminsky) [inline helicopter.py]: do not invoke \_\_dict\_\_ more than necessary. Make variable attr_dict = self.\_\_dict\_\_ and then use it two times in comprehension

**Review by MarkoYaminsky — APPROVED** [review]
good boy

**Pull request merged**